tests/test_tag.py

```diff
@@ -155,3 +156,37 @@ def test_push_tag_failed(mock_repo):
     with patch('git.repo.fun.name_to_object'):
         with pytest.raises(exceptions.PushException):
             repo.tag.push("my_tag", "origin")
+
+
+def test_list_tags(mock_repo):
+    """
+   GIVEN GitRepo is initialized with a path and repo
+   WHEN GitTag.list_tags() is called
+   THEN repo.tags is returned
+   """
+    expected = ['tag1', 'tag2']
+    my_local_mock = mock_repo
+    type(my_local_mock).tags = PropertyMock(return_value=expected)
+    repo = GitRepo(repo=my_local_mock)
+
+    test_tags = GitTag(repo, None).list_tags()
+
+    assert expected == test_tags
+
+
+def test_list_tags_failed(mock_repo):
+    """
+   GIVEN GitRepo is initialized with a path and repo
+   WHEN GitTag.list_tags() is called
+   AND repo.tags fails
+   THEN a TaggingException is raised
+   """
+    def func():
+        raise git.GitCommandError('tags', '')
+
+    my_local_mock = mock_repo
+    type(my_local_mock).tags = PropertyMock(side_effect=func)
```

Comment:
I'm just as suspicious about type() here, if the tests and mocking are still causing issues I can try and have a look once the code is updated, but this doesn't look right to me.

Reply:
See the above comment on type usage with PropertyMock.